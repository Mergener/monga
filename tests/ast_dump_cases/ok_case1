function hello(a : int, b : float) {
	a = 5;
	a = a;
}