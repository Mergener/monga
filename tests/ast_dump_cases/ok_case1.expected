<Module>
	<FunctionDefinition>
		<Name>hello</Name>
		<ReturnType>void</ReturnType>
		<Parameters>
			<Parameter>
				<Name>a</Name>
				<Type>int</Type>
			</Parameter>
			<Parameter>
				<Name>b</Name>
				<Type>float</Type>
			</Parameter>
		</Parameters>
		<Block>
			<Assignment>
				<Var>a</Var>
				<IntConstant>5</IntConstant>
			</Assignment>
			<Assignment>
				<Var>a</Var>
				<Var>a</Var>
			</Assignment>
		</Block>
	</FunctionDefinition>
</Module>
